packages/core/src/hydration/annotate.ts

```diff
@@ -52,6 +52,7 @@ class SerializedViewCollection {
  */
 interface HydrationContext {
   serializedViewCollection: SerializedViewCollection;
+  corruptedTextNodes: Map<HTMLElement, string>;
```

Comment:
I think it'd be great to add a JSDoc comment above this to explain what the map is used for (a couple of sentences).

Reply:
We can also use an enum to better represent the operation and its type:
```suggestion
  corruptedTextNodes: Map<HTMLElement, TextNodeMarker>;
```
where `TextNodeMarker` may look like this:
```
export const enum TextNodeMarker {
  EmptyNode = 'ngetn';
  Separator = 'ngtns';
}
```

Reply:
This is great idea. I didn't even think about an enum.